File: be/src/main/java/sg/flow/services/AccountServices/AccountService.java
package sg.flow.services.AccountServices;

import java.util.List;

import sg.flow.models.account.AccountWithTransactionHistory;
import sg.flow.models.account.BriefAccount;

public interface AccountService {

    List<BriefAccount> getBriefAccounts(Integer userId);

    List<AccountWithTransactionHistory> getAccountWithTransactionHistorys(Integer userId);

    BriefAccount getBriefAccount(Integer userId, Long accountId);

    AccountWithTransactionHistory getAccountWithTransactionHistory(Integer userId, Long accountId);

}

File: be/src/main/kotlin/sg/flow/services/AccountServices/AccountService.kt
package sg.flow.services.AccountServices

import sg.flow.models.account.AccountWithTransactionHistory
import sg.flow.models.account.BriefAccount


interface AccountService {
suspend fun getAccounts(userId: Int): List<BriefAccount>

    suspend fun getAccountWithTransactionHistory(userId: Int): List<AccountWithTransactionHistory>
suspend fun getAccount(userId: Int, accountId: Long): BriefAccount

    suspend fun getAccountWithTransactionHistory(
            userId: Int,
            accountId: Long
    ): AccountWithTransactionHistory
}
